### ISSUE TITLE
Casting variable positional argument yields 'No Value for argument' error.

### ISSUE THREAD

**Issue by mlfz** [PyCQA/pylint#878]
pylint flagged the following code
```py
def foo(a):
    return a

def bar(*args):
    foo(*args) # okay
    foo(*list(args)) # no value for argument error

```
with errors:
```
$ pylint -E foo.py 
************* Module foo
  6, 4: E1120: bar: No value for argument 'a' in function call
```

This is triggering false positives for functions that acts as wrapper for simple arguments manipulator before calling another function.

```
$ pylint --version
pylint 1.5.3, 
astroid 1.4.3
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2]
```

**Comment by PCManticore** [comment]
I can reproduce it, thanks for reporting the issue!
